source/slang/parameter-binding.cpp

```diff
@@ -443,6 +443,36 @@ static void splitNameAndIndex(
     outDigits = UnownedStringSlice(digitsBegin, digitsEnd);
 }
 
+LayoutResourceKind findRegisterClassFromName(UnownedStringSlice const& registerClassName)
+{
+    switch( registerClassName.size() )
+    {
+    case 1:
+        switch (*registerClassName.begin())
```

Comment:
I'm not sure the `UnownedStringSlice` type has an `operator[]`, because the code that I was adapting used `*registerClassName.begin()` already (and this case is ultimately just a shuffled-around version of the code that was there before, to accommodate the new case).